presto-hive/src/main/java/com/facebook/presto/hive/HiveFileSplit.java

```diff
@@ -24,6 +24,11 @@
 import static com.google.common.base.Preconditions.checkArgument;
 import static java.util.Objects.requireNonNull;
 
+/**
+ * Represents a piece of a file at a given path starting at a given position and continuing for length bytes.
+ * All numeric values must be zero or greater. Object values must be non-null.
```

Comment:
Suggest removing this line. I found it hard to understand what you meant here (i thought you were saying all numeric values in the file had to be non-negative), it can easily get out of date if someone happens to add some other field that can be negative, and this information is pretty easy to find in the constructor.  If you don't want to remove it I suggest specifying which fields can't be negative.  
```suggestion
```

Reply:
done